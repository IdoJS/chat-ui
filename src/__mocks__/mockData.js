const MOCK_USER_NAME = 'mock_user_name';
const MOCK_AVATAR_CLASS = 'mock_avatar_class';
const MOCK_TEXT = 'mock_text';
const MOCK_SOCKET_REQ = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, userId: '0'};
const MOCK_SELF_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, timestamp: 0, userId: '0'};
const MOCK_OTHER_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: 'other_user', timestamp: 0, userId: '1'};

const MOCK_CHAT_DATA_INIT = {
  loading: false,
  timestamp: Date.now(),
  chatArr: [],
  typingArr: [],
  timeout: false
};
export {
  MOCK_USER_NAME,
  MOCK_AVATAR_CLASS,
  MOCK_TEXT,
  MOCK_SOCKET_REQ,
  MOCK_CHAT_DATA_INIT,
  MOCK_SELF_MESSAGE,
  MOCK_OTHER_MESSAGE
}