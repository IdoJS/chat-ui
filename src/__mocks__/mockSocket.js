// modify the code https://medium.freecodecamp.org/testing-socket-io-client-app-using-jest-and-react-testing-library-9cae93c070a3

let EVENTS = {};
function emit(event, ...args) {
  EVENTS[event].forEach(func => func(...args));
}
const socket = () => {
  return {
    on(event, func) {
      if (EVENTS[event]) {
        return EVENTS[event].push(func);
      }
      EVENTS[event] = [func];
    },
    emit
  }
};
export const mockIo = {
  connect() {
    return socket;
  }
};

export default mockIo;