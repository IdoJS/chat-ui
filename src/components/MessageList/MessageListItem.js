import React from 'react';
import PropTypes from 'prop-types';
import {splitTimestamp} from '../../utils/formats';

const MessageListItem = (props) => {
  const {data, currentUserId} = props;
  const {text, userName, timestamp, avatar, userId} = data;
  const currentTimeArr = splitTimestamp(timestamp);
  const isCurrentUserMsg = userId === currentUserId;

  return <li className='chat_item'>
    <div className={`chat_item_text_header chat_item${isCurrentUserMsg ? '_user' : '_default'}`}>
      <h3 className='chat_item_user_name'>{userName}</h3>
      <h3>{currentTimeArr[1]}</h3>
    </div>
    <div className={`ui message chat_item_msg chat_item_msg${isCurrentUserMsg ? '_user' : '_default'}`}>
      <div></div>
      <div className={`chat_item_img avatar ${avatar}`}/>
      <div className={`chat_item_text chat_item_msg chat_item_msg${isCurrentUserMsg ? '_user' : '_default'}`}>
        <p
          className={`ui ${isCurrentUserMsg ? 'left pointing blue' : 'right pointing teal'} chat_item_msg${isCurrentUserMsg ? '_user' : '_default'} basic label`}>{text}</p>
      </div>
    </div>
  </li>
};

MessageListItem.propTypes = {
  currentUser: PropTypes.string.isRequired,
  currentUserId: PropTypes.string.isRequired,
  data: PropTypes.shape({
    text: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired
  }).isRequired
};

MessageListItem.defaultProps = {
  currentUser: '',
  data: {
    text: '',
    userName: '',
    timestamp: Date.now()
  }
};

export default MessageListItem;


