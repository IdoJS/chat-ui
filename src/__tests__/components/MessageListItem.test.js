import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {MOCK_USER_NAME, MOCK_SELF_MESSAGE, MOCK_OTHER_MESSAGE} from '../../__mocks__/mockData';

import MessageListItem from '../../../src/components/MessageList/MessageListItem';

describe('MessageListItem component', () => {
  it('Should render the MessageListItem view', () => {

    const wrapper = shallow(<MessageListItem currentUser={MOCK_USER_NAME} data={MOCK_SELF_MESSAGE}/>);
    const p = wrapper.find('p');

    expect(p.text()).toEqual(MOCK_SELF_MESSAGE.text);
    expect(wrapper.hasClass('my_msg')).toBeTruthy();

  });

  it('Should render the MessageListItem view', () => {

    const wrapper = shallow(<MessageListItem currentUser={MOCK_USER_NAME} data={MOCK_OTHER_MESSAGE}/>);
    const p = wrapper.find('p');

    expect(p.text()).toEqual(MOCK_OTHER_MESSAGE.text);
    expect(wrapper.hasClass('default_msg')).toBeTruthy();

  });

  it('SnapShot rendering MessageListItem', () => {
    const wrapper = shallow(<MessageListItem currentUser={MOCK_USER_NAME} data={MOCK_SELF_MESSAGE}/>);
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

});