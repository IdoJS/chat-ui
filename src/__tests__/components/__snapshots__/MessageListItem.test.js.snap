// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageListItem component SnapShot rendering MessageListItem 1`] = `
<li
  className="chat_item my_msg"
>
  <div
    className="chat_item_avatar"
  >
    <div
      className="chat_item_avatarImg mock_avatar_class"
    />
    <h3
      className="chat_item_timestamp "
    >
      <div>
        00:00:00
      </div>
      <div>
        1970-01-01
      </div>
    </h3>
  </div>
  <p
    className="chat_item_text my_msg"
  >
    mock_text
  </p>
</li>
`;
